dmoj/cptbox/sandbox.py

```diff
@@ -328,7 +328,7 @@ def __init_streams(self, stdin, stdout, stderr):
 
         if stdin == PIPE:
             self._child_stdin, self._stdin = os.pipe()
-            self.stdin = os.fdopen(self._stdin, 'w')
+            self.stdin = os.fdopen(self._stdin, 'wb')
```

Comment:
Passing the binary flag causes the writers to change from `_io.TextIOWrapper` to `_io.BufferedWriter`, which does not actually write the data...

Reply:
Huh? Everyone knows that if you don't want buffering, pass `buffering=0`.